packages/core/src/modules/credentials/CredentialsModuleConfig.ts

```diff
@@ -0,0 +1,21 @@
+import { AutoAcceptCredential } from './models'
+
+/**
+ * CredentialsModuleConfigOptions defines the interface for the options of the CredentialsModuleConfig class.
+ * This can contain optional parameters that have default values in the config class itself.
+ */
+export interface CredentialsModuleConfigOptions {
+  autoAcceptCredentials: AutoAcceptCredential
+}
+
+export class CredentialsModuleConfig {
```

Comment:
This is awesome! Way more clear which properties from the agent config are with that module. 

Just more curious, how do we deal with configuration shared between modules, if it ever occurs?

Reply:
Good question. I think either declare it in both modules and provide it twice, or one module can depend on the configuration of the other module (if they are dependant on each other).

E.g. the outOfBandModule can depend on the `autoAcceptConnections` from the `ConnectionsModule` (as you can't use the oob module without the connections module). Does that make sense?

Reply:
Yeah that makes sense. Is it obvious to the user that the `autoAcceptConnections` from the `ConnectionsModule` will be used in the `outOfBandModule`?

Reply:
Not sure. We could add it to the documentation of the out of band module. So in the receiveInvitation method documentation we can mention it will use the `autoAcceptConnections` property from the connections module config to determine whether to auto accept the config. 

I've added the config of each module as a public readonly property to the api class, which means it's public api and makes sense it can be used by other modules.

So `agent.connections.config` is a public property that exposes the config of the connections module